src/serializer/index.ts

```diff
@@ -23,22 +27,51 @@ export class Serializer<
 			: R
 		: FirstOut,
 > {
+	private queue = Symbol('serializerQueue');
+	private readonly options: SerializerOptions;
 	private readonly chain: Chain;
 	private readonly lastChain: number;
-	constructor(private strategy: MainStrategy, ...chain: Chain) {
-		this.chain = chain;
-		this.lastChain = chain.length - 1;
+
+	constructor(strategy: MainStrategy, ...chain: Chain);
+	constructor(
+		strategy: MainStrategy,
+		options: SerializerOptions,
+		...chain: Chain
+	);
+	constructor(
```

Comment:
The option is informed through constructor overload, to maintain compatibility with previous behavior